src/github-installer.ts

```diff
@@ -0,0 +1,143 @@
+        opaOutputChannel.appendLine(`Executable downloaded to ${path}`);
+    } catch (e) {
+        opaOutputChannel.appendLine('Something went wrong while downloading the executable:');
+        opaOutputChannel.appendLine(e as string);
+        return;
+    }
+
+    opaOutputChannel.appendLine('Changing file mode to 0755 to allow execution...');
+    try {
+        fs.chmodSync(path, 0o755);
+    } catch (e) {
+        opaOutputChannel.appendLine(e as string);
+        return;
+    }
+
+    opaOutputChannel.appendLine(`Setting 'opa.dependency_paths.${binaryConfigKey}' to '${path}'...`);
+    try {
+        await vscode.workspace.getConfiguration('opa.dependency_paths').update(binaryConfigKey, path, true);
+    } catch (e) {
+        opaOutputChannel.appendLine('Something went wrong while saving the config setting:');
+        opaOutputChannel.appendLine(e as string);
+        return;
+    }
+    opaOutputChannel.appendLine(`Successfully installed ${repo}!`);
+    opaOutputChannel.hide();
+}
+
+// Downloads a file given a URL
+// Returns a Promise that resolves to the absolute file path when the download is complete
+async function downloadFile(url: URL, determineExecutableName: () => string): Promise<string> {
+    // Use the user's home directory as the default base directory
+    const dest = os.homedir();
+    return releaseDownloader.downloadAsset(url.href, determineExecutableName(), dest, () => {
+        // Basic progress bar
+        if (Math.floor(Date.now()).toString().endsWith('0')) {
+            opaOutputChannel.append('.');
+        }
+    });
+}
+
+async function getDownloadUrl(repo: string, determineBinaryURLFromRelease: (release: any) => string): Promise<URL> {
+    return new Promise<URL>((resolve, reject) => {
+        // TODO: Honor HTTP proxy settings from `vscode.workspace.getConfiguration('http').get('proxy')`
+        https.get({
+            hostname: 'api.github.com',
+            path: `/repos/${repo}/releases/latest`,
+            headers: {
+                'User-Agent': 'node.js',
+                'Authorization': `token ${getToken()}`
+            }
+        }, (res: http.IncomingMessage) => {
+            let rawData = '';
+            res.on('data', (d: any) => {
+                rawData += d;
+            });
+            res.on('end', () => {
+                try {
+                    const release = JSON.parse(rawData);
+                    const url = determineBinaryURLFromRelease(release);
+                    resolve(new URL(url));
+                } catch (e) {
+                    reject(e);
+                }
+            });
+        }).on('error', (e: any) => reject(e));
+    });
+}
+
+function getToken(): string {
+    // Need an OAuth token to access Github because
+    // this gets around the ridiculously low
+    // anonymous access rate limits (60 requests/sec/IP)
+    // This token only gives access to "public_repo" and "repo:status" scopes
```

Comment:
This comment certainly is needed! 😄

Reply:
Ha indeed, it's copied from the old implementation however and so isn't new to this PR.